app/lib/features/chat/widgets/text_message_builder.dart

```diff
@@ -184,4 +186,38 @@ class _TextWidget extends ConsumerWidget {
       ],
     );
   }
+
+  Future<void> onLinkTap(Uri uri, BuildContext context, WidgetRef ref) async {
+    final roomId = getRoomIdFromLink(uri);
+
+    //If link is type of matrix room link
+    if (roomId != null) {
+      joinAndOrNavigateToChatRoom(context, ref, roomId);
```

Comment:
So, clicking on link directly attempts to join to the room automatically, this doesn't look good. Ideally I would want some consensus for user here .i.e. showing confirmation or alert dialog prior to join room action.

Reply:
Updated code and reference video in the description area of the PR. :)